### PR TITLE
netstandard2

### PR DESCRIPTION
**Pull request by jchannon**
Hi there!

Thank you so much for contributing this pull request. We love when the community
steps up and share, and we're enormously thankful.

You don't have to do all of those following things, they are what we try and get 
done for code coming in:

 - [x] the code
 - [ ] If it's a non-trivial piece of code, signing a CLA
 - [ ] If it breaks, change, fix or add to existing behaviour, updating
       CHANGELOG.md
 - [ ] If it's a bug fix, tests in the Tests `project`, or scenarios in the `TestRig`.
 - [ ] On top of HEAD, unless it's a bugfix on a previous version
 - [ ] VERSION file updated if not creating a pull-request on top of `master`

If those are needed, and you haven't and can't, we thank you enromously for your
contirbution, and we'll add those before merging the pull request.

Thanks!

The OpenRasta community

### CODE REVIEW

**Comment by jchannon** [comment]
So there's 3 areas to highlight:

i) TraceSourceLogger - it uses a property System.Diagnostics.Debug.Listeners which is not in netstandard2 but in net461
ii) AppDomainHost - a few issues around AppDomain which is not in netstandard2 but in net461
iii) ExtensibleCSharpCodeProvider - this inherits off CSharpCodeProvider which is in package System.CodeDom for netstandard2

I've added conditional framework checks in csproj and constants to #ifdef compilation based on whether you can target net461 or netstandard2

**Comment by jchannon** [comment]
Here's a report for portability. In a lot of cases it usually recommends what APIs changed and how you can use newer ones to replicate functionality. Here we're not so lucky

screen shot 2017-12-19 at 09 23 02

**Comment by serialseb** [comment]
I think we can happily code conditionl everything that's not supported, most of these are not used anymore.

The compiler extensions should just be deleted, no one has used that in the last 10 years

**Comment by jchannon** [comment]
Do you want to merge this and then go through and define what needs
deleting or do that as part of this PR?

**Comment by serialseb** [comment]
Could you look at the appveyor build?

**Comment by jchannon** [comment]
Yup will do 😄

**Comment by jchannon** [comment]
Thoughts @serialseb ?

File: src/OpenRasta.Tests.Unit/Web/MediaType_Specification.cs [hunk 1]
```diff
@@ -40,13 +40,14 @@ public void no_valid_value_throws()
...
     }
-
+#if !CORE
     [Test, SetCulture("fr-FR")]
     public void parsing_a_quality_value_when_using_another_culture_still_parse_the_dot_value()
     {
       var mediaType = new MediaType("application/xml;q=0.3");
       mediaType.Quality.ShouldBe(0.3f);
     }
+#endif
```
Comment (serialseb) [inline src/OpenRasta.Tests.Unit/Web/MediaType_Specification.cs]: What's the reason for that?

**Comment by serialseb** [comment]
Should we update the CHANGELOG too to specify we target netstandard2 and 461 now?

File: src/OpenRasta.Tests.Unit/Web/MediaType_Specification.cs (hunk 1 above)
Comment (jchannon) [reply src/OpenRasta.Tests.Unit/Web/MediaType_Specification.cs]: NUnit doesn't support SetCulture on netcore yet but planned - https://github.com/nunit/nunit/issues/1696

**Comment by jchannon** [comment]
Happy to update CHANGELOG but can't quite determine what bit to change, last date seems to be `[2.5.1050] - 2016-12-15`

**Pull request merged**